tox.ini

```diff
@@ -1,17 +1,36 @@
 [tox]
-envlist = py27
+envlist = py{35,36,37}
 
-[testenv]
-usedevelop = true
+[base]
 deps =
-    -rrequirements.txt
     pyftpdlib
     parameterized
     pytest-xdist
     pytest-cov
-    miniboa
+    miniboa >= 1.0.8
     biplist
+
+[testenv]
+usedevelop = true
+deps =
+    -rrequirements.txt
+    {[base]deps}
 commands =
     py.test {posargs:--tb=short --cov=linkcheck tests}
 setenv =
     LC_ALL=en_US.utf-8
+
+[testenv:py35]
+# test bs4 without line number support
+usedevelop = true
+deps =
+    beautifulsoup4 < 4.8.1
+    requests >= 2.4
+    pyxdg
+    dnspython
+    future
+    {[base]deps}
+commands =
+    {[testenv]commands}
+setenv =
+    {[testenv]setenv}
```

Comment:
commands/setenv/usevelop will be inherited from [testenv] by default so you don't need to specify those again.

(And if you go with my suggestion to use factors, you won't need a separate [testenv:py35] section either.)

Reply:
Removed redundant commands and setenv as suggested.